File: P2/NapakalakiRuby/lib/dice.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Dice
  include Singleton
  att_reader :instance
 
  
  def initialize
    @@instance=nil
  end
    
  def nextNumber
    
  end
end

File: P4/Napakalaki/src/napakalaki/Dice.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.Random;
/**
 *
 * @author Victor Vallecillo Morilla
 */
public class Dice
{
    private static Dice instance = null;
    
    private Dice()
    {      
    }
    public static Dice getInstance() {
        if (instance == null) {
            instance = new Dice();
        }
        return instance;
    }
    
    public int nextNumber()
    {
        Random m=new Random();
        
        return(m.nextInt((6)+1));
    }
    
}
